code/game/objects/items/devices/flashlight.dm

```diff
@@ -206,3 +229,79 @@
 
 /obj/item/device/flashlight/slime/attack_self(mob/user)
 	return //Bio-luminescence does not toggle.
+
+//Glowsticks
+
+/obj/item/device/flashlight/glowstick
+	name = "green glowstick"
+	desc = "A green military-grade glowstick."
+	w_class = 2
+	brightness_on = 4
+	light_power = 2
+	light_color = "#49F37C"
+	icon = 'icons/obj/glowsticks.dmi'
+	icon_state = "glowstick"
+	item_state = "glowstick"
+	contained_sprite = 1
+	offset_light = 0
+	diona_restricted_light = 0
+	var/fuel = 0
+
+/obj/item/device/flashlight/glowstick/New()
+	fuel = rand(1200, 1700)

```

Comment:
This duration is a minimum of 40 minutes, seems quite long. how easy are these to obtain?